confluence-syntax-xhtml/src/main/java/org/xwiki/contrib/confluence/parser/xhtml/internal/wikimodel/TimeTagHandler.java

```diff
@@ -34,20 +33,34 @@
  * @version $Id$
  * @since 9.5
  */
-public class TimeTagHandler extends TagHandler implements ConfluenceTagHandler
+public class TimeTagHandler extends MacroTagHandler
 {
     /**
      * Constructor (checkstyle complains that this comment is missing although the other classes don't have it).
      */
     public TimeTagHandler()
     {
-        super(false);
+        super();
     }
 
     @Override
     protected void begin(TagContext context)
     {
-        String date = context.getParams().getParameter("datetime").getValue();
-        context.getScannerContext().onWord(date);
+        // Associate the time tag to a "date" macro that has a "value" and a "format" parameter.
+        ConfluenceMacro macro = new ConfluenceMacro();
+
+        macro.name = "date";
```

Comment:
Why "date" ? Since the Confluence element name is "time", that's what I would expect to end up with since the point is to produce a Confluence place holder.

Reply:
We don't have any equivalent `time` or `date` macro in XWiki, but I'll add one in the Task Manager Application. 

The naming convention was used to better reflect what the macro represents, a date. Naming it "time" one could think of it as a macro that would display the time of the day, some hour and minute, when it in fact should display a date. (as it does in Confluence)

The "value" parameter simply replaces the "datetime" parameter of the confluence tag. The "format" parameter is used to determine the format of the "datetime"/"value" parameter so we will be able to parse it. The hard-coded format I introduced was extracted from the exported confluence spaces and instances. It seems that they use this format to store their data.

Ultimately, the goal was to make the following transformation.

`<time datetime='2023-02-24' />` -> `{{date value='2023-02-24' format='yyyy-MM-dd'/}}`

Reply:
> The naming convention was used to better reflect what the macro represents, a date. Naming it "time" one could think of it as a macro that would display the time of the day, some hour and minute, when it in fact should display a date. (as it does in Confluence)

It's not really our problem if Confluence chose this naming. Also, are you 100% sure the `time` element never support anything else than a date ? There must be a reason why they named the parameter datetime, maybe there is a way to change the format used in that parameter.

I really feel it's asking for troubles to assume what you see in one example is the specification, it would make a lot more sense at the syntax parser level to keep Confluence naming. You should move your conversion logic to a converter since that's what they are for.

Reply:
There's no standard XWiki ``time`` macro either. For me the question is more: if we were to add a standard XWiki macro to display a date & time value how would we name it? ``time`` is certainly not the first option that comes to my mind. I'd name it ``date`` or ``dateTime``. And for this hypothetical "date" macro would we expect a hard-coded date & time format, i.e. would we force the users to input the date & time value using always the same format? Certainly not. It makes more sense to be able to specify the date & time format used to parse the date & time value, as a macro parameter, besides the value. This leads to:

    {{date value="2023-02-24" format="yyyy-MM-dd" /}}
    {{date value="24/02/2023" format="dd/MM/yyyy" /}}

I wouldn't preserve the Confluence macro name and parameters if they are not good enough (not generic enough). If you don't agree with this then we'll have to:

* [ideally] namespace or prefix the generated macro name to clearly mark the fact that it comes from a Confluence import:
  ```
  {{cf:time datetime="2023-02-24" /}}
  ```
* implement this Confluence macro as a _bridge_ macro, bridging it to the more generic ``date`` macro @trrenty is implementing for the Task Manager application, using the date & time format specific to Confluence (``yyyy-MM-dd``):
  ```
  => {{date value="2023-02-24" format="yyyy-MM-dd" /}}
  ```

In any case, the main reason @trrenty proposed using a ``date`` macro is because he's implementing this macro in the Task Manager application for now, but it's a good candidate for a standard XWiki macro.

Reply:
> [ideally] namespace or prefix the generated macro name to clearly mark the fact that it comes from a Confluence import: 

That's filter's job (and it's using `confluence_` prefix by default).

You are mixing the low level syntax parser (which is where we are here and which should produce events corresponding to the parsed content) and the filter, which, among other things, do conversions.

The final result will go through both, and again I don't mind moving that conversion to a converter since that's what they are for.

Reply:
Thank you for your explanation Thomas. I don't know the Confluence importer internals. I was talking about the end result. I'm fine to produce a ``time`` macro here, as it is in Confluence, and to convert it to a ``date`` macro in a filter. I suppose this filter could be provided by the Task Manager Application (in a dedicated optional module) which implements the ``date`` macro ATM.

Reply:
> I suppose this filter could be provided by the Task Manager Application (in a dedicated optional module) since that's where the date macro is implemented ATM.

Yes, it would also make sense for the Confluence macro converter to be provided by the Task Manager project, since that's the one which know about the date macro. Basically, it all depends on if we prefer the task manager project to know about Confluence or the Confluence filter project to know about the Task manager date macro. It could also be another project, but I feel it's overkill.

Since Task Manager Pro (I guess that's the one you have in mind and not the free one) is not free, I guess it make more sense to have the converter in it as it would not make much sense for Confluence to convert to produce something that only make sense with a paying macro.

Reply:
I changed the code accordingly. Now the confluence time tag `<time datetime="1/11/1111" />` is converted to a macro call. `{{time datetime="1/11/1111" /}}`. The XWiki Converter class was added to the Task Application project.

Reply:
Thanks @trrenty !